lib/core/src/irods_hierarchy_parser.cpp

```diff
@@ -87,10 +87,29 @@ namespace irods {
 
     error hierarchy_parser::add_child(
         const std::string& _resc ) {
+        if (_resc == hierarchy_parser::delimiter()) {
+            return ERROR(SYS_INVALID_INPUT_PARAM, "invalid resource name");
+        }
 
-        error result = SUCCESS();
         resc_list_.push_back( _resc );
-        return result;
+        return SUCCESS();
+    }
+
+    void hierarchy_parser::add_parent(
+        const std::string& _parent,
+        const std::string& _child) {
+        if (_parent == hierarchy_parser::delimiter()) {
+            THROW(SYS_INVALID_INPUT_PARAM, "invalid resource name");
+        }
+        auto it = resc_list_.begin();
+        if (!_child.empty()) {
+            it = std::find(resc_list_.begin(), resc_list_.end(), _child);
+            if (it == resc_list_.end()) {
+                THROW(CHILD_NOT_FOUND, (boost::format(
```

Comment:
Need to change this to use `fmt::format`... been sitting on this commit for quite some time!